roles/openshift_hosted/tasks/router/router.yml

```diff
@@ -11,69 +11,34 @@
 - name: set_fact replicas
   set_fact:
     replicas: "{{ openshift.hosted.router.replicas|default(None) | get_router_replicas(router_nodes) }}"
+    openshift_hosted_router_selector: "{{ openshift.hosted.router.selector | default(None) }}"
+    openshift_hosted_router_image: "{{ openshift.hosted.router.registryurl }}"
 
-- block:
-
-  - name: Assert that 'certfile', 'keyfile' and 'cafile' keys provided in openshift_hosted_router_certificate
-    assert:
-      that:
-      - "'certfile' in openshift_hosted_router_certificate"
-      - "'keyfile' in openshift_hosted_router_certificate"
-      - "'cafile' in openshift_hosted_router_certificate"
-      msg: "'certfile', 'keyfile' and 'cafile' keys must be specified when supplying the openshift_hosted_router_certificate variable."
-
-  - name: Read router certificate and key
-    become: no
-    local_action:
-      module: slurp
-      src: "{{ item }}"
-    register: openshift_router_certificate_output
-    # Defaulting dictionary keys to none to avoid deprecation warnings
-    # (future fatal errors) during template evaluation. Dictionary keys
-    # won't be accessed unless openshift_hosted_router_certificate is
-    # defined and has all keys (certfile, keyfile, cafile) which we
-    # check above.
-    with_items:
-    - "{{ (openshift_hosted_router_certificate | default({'certfile':none})).certfile }}"
-    - "{{ (openshift_hosted_router_certificate | default({'keyfile':none})).keyfile }}"
-    - "{{ (openshift_hosted_router_certificate | default({'cafile':none})).cafile }}"
-
-  - name: Persist certificate contents
-    openshift_facts:
-      role: hosted
-      openshift_env:
-        openshift_hosted_router_certificate_contents: "{% for certificate in openshift_router_certificate_output.results -%}{{ certificate.content | b64decode }}{% endfor -%}"
-
-  - name: Create PEM certificate
-    copy:
-      content: "{{ openshift.hosted.router.certificate.contents }}"
-      dest: "{{ openshift_master_config_dir }}/openshift-router.pem"
-      mode: 0600
-
-  when: openshift_hosted_router_certificate is defined
+- name: Get the certificate contents for router
+  copy:
+    backup: True
+    dest: "/etc/origin/master/{{ item | basename }}"
```

Comment:
@abutcher seems like it would be better to either:
- Only do the copy if the contents differ from what is already configured
- Pass the certificate contents into oc_adm_router, with the content coming from either a variable or a local file lookup on the control host (my preference here would be to support both).